File: frontend/src/app/core/services/caja.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CajaResponse, CajaAperturaRequest, CajaCierreRequest } from '../../models/caja.model';
import { VentaResponse } from '../../models/venta.model';

@Injectable({ providedIn: 'root' })
export class CajaService {

  private readonly API_URL = '/api/caja';

  constructor(private http: HttpClient) {}

  aperturar(dto: CajaAperturaRequest): Observable<CajaResponse> {
    return this.http.post<CajaResponse>(`${this.API_URL}/aperturar`, dto);
  }

  cerrar(id: number, dto: CajaCierreRequest): Observable<CajaResponse> {
    return this.http.post<CajaResponse>(`${this.API_URL}/${id}/cerrar`, dto);
  }

  obtenerActiva(): Observable<CajaResponse | null> {
    return this.http.get<CajaResponse | null>(`${this.API_URL}/activa`);
  }

  listarMias(): Observable<CajaResponse[]> {
    return this.http.get<CajaResponse[]>(`${this.API_URL}/mias`);
  }

  listarTodas(): Observable<CajaResponse[]> {
    return this.http.get<CajaResponse[]>(this.API_URL);
  }

  obtenerPorId(id: number): Observable<CajaResponse> {
    return this.http.get<CajaResponse>(`${this.API_URL}/${id}`);
  }

  obtenerVentas(id: number): Observable<VentaResponse[]> {
    return this.http.get<VentaResponse[]>(`${this.API_URL}/${id}/ventas`);
  }
}

File: backend/src/main/java/com/kineticrehab/service/CajaService.java
package com.kineticrehab.service;

import com.kineticrehab.dto.request.CajaAperturaRequestDTO;
import com.kineticrehab.dto.request.CajaCierreRequestDTO;
import com.kineticrehab.dto.response.CajaResponseDTO;
import com.kineticrehab.dto.response.VentaResponseDTO;

import java.util.List;

public interface CajaService {

    CajaResponseDTO aperturar(CajaAperturaRequestDTO dto);

    CajaResponseDTO cerrar(Long id, CajaCierreRequestDTO dto);

    CajaResponseDTO obtenerActiva();

    List<CajaResponseDTO> listarMias();

    List<CajaResponseDTO> listarTodas();

    CajaResponseDTO obtenerPorId(Long id);

    List<VentaResponseDTO> obtenerVentas(Long cajaId);
}
